Mobile Buy SDK/Mobile Buy SDK/Utils/BUYApplePayAuthorizationDelegate.m

```diff
@@ -177,9 +174,39 @@ - (void)updateCheckoutWithAddressCompletion:(void (^)(PKPaymentAuthorizationStat
 		[self.client updateCheckout:self.checkout completion:^(BUYCheckout *checkout, NSError *error) {
 			if (checkout && error == nil) {
 				self.checkout = checkout;
-				[self getShippingRates:self.checkout completion:completion];
-			}
-			else {
+				
+				if ([self.checkout requiresShipping] == NO) {
+					completion(PKPaymentAuthorizationStatusSuccess, nil, [self.checkout buy_summaryItemsWithShopName:self.shopName]);
+				} else {
+					
+					self.shippingRates = @[];
+					[self.client getShippingRatesForCheckoutWithToken:self.checkout.token completion:^(NSArray *shippingRates, BUYStatus status, NSError *error) {
+						
+						self.shippingRates = shippingRates;
+						if (shippingRates) {
+							
+							NSArray *shippingMethods = [BUYShippingRate buy_convertShippingRatesToShippingMethods:shippingRates];
+							if (shippingMethods.count > 0) {
+								
+								[self selectShippingMethod:shippingMethods[0] completion:^(BUYCheckout *checkout, NSError *error) {
```

Comment:
We should not need to pass completion blocks defined by self into methods owned by self. The `-selectShippingMethod:` method should be able to create this block, and will be able to if the caller's original completion block was available as an ivar or property.